lib/models/api_calls.dart

```
@@ -0,0 +1,92 @@
+
+class ApiCalls {
+  ///Url to get the intial page of the list
+  static const DEVICE_LIST_URL =
+      'https://www.googleapis.com/admin/directory/v1/customer/my_customer/devices/chromeos?projection=BASIC&maxResults=15';
+
+  ///Url to get the next page of the list of devices with a next page token
+  static const DEVICE_LIST_WITH_TOKEN_URL =
+      'https://www.googleapis.com/admin/directory/v1/customer/my_customer/devices/chromeos?projection=BASIC&maxResults=15&pageToken=';
+
+  ///Url to get detailed information of a device
+  static const DETAILED_DEVICE_URL =
+      'https://www.googleapis.com/admin/directory/v1/customer/my_customer/devices/chromeos/';
+
+  /// Calls the Directory API to get a list of devices with max results of 15.
+  ///
+  /// * [client] is the http client that will do the https requests.
+  /// * [authToken] is the authentication token for API to accept the requests.
+  /// * [nextPageToken]  is the next page of the devices list that should
+  /// retreive, if it is null the function will request the first page.
+  ///
+  /// This method returns a [AccountDevices] object.
+  static getDeviceList(
+      http.Client client, String authToken, String nextPageToken) async {
+    var response;
+    authToken = 'Bearer $authToken';
+    if (nextPageToken == null) {
+      response = await client.get(DEVICE_LIST_URL,
+          headers: {HttpHeaders.authorizationHeader: authToken});
+    } else {
+      response = await client.get(DEVICE_LIST_WITH_TOKEN_URL + nextPageToken,
+          headers: {HttpHeaders.authorizationHeader: authToken});
+    }
+    // If conection is succesful return a AccountDevices object by parsing the
+    // response body
+    if (response.statusCode == 200) {
+      return AccountDevices.fromJson(json.decode(response.body));
+    } else {
+      // If that call was not successful, throw an error.
+      throw Exception('${response.statusCode} ${response.body}');
```

If we are planning to show different experiences based on the status code, maybe we should throw the custom exception (or rethrow http error) with the status code, so the controller can differentiate what to do with each type of error

---

Sure I'm passing the status code, so I can handle it latter depending of the code.